pkg/runtime/setup.go

```diff
@@ -269,11 +281,18 @@ func (s *setup) update() error {
 		}
 	}
 
+	// Tell applicable ecosystems to perform any final uninstall actions.
```

Comment:
I'm not convinced we can have `Apply()` do both uninstall and install at the same time, so I'm calling `Apply()` here after all of the `Remove()` calls, and then again later in this file after all of the `Add()` calls.